org.eclipse.dawnsci.hdf5.model.test/src/org/eclipse/dawnsci/hdf5/model/internal/HierarchicalDataFileModelTest.java

```diff
@@ -306,35 +319,38 @@ public void testReadDatasetFirst() {
 				model.getPath("/entry1/instrument/analyser/cps@units"));
 	}
 
-	private Object createNexusFile(final int dtype, final long[] shape, final Object buffer) throws Exception {
+	private Object createNexusFile(final int dtype, final int[] shape, final Object buffer) throws Exception {
 		
-		String PATH = "kichwa1";
+		String PATH = Node.SEPARATOR + "kichwa1";
+		String dataPath = PATH + Node.SEPARATOR + "Gold";
 		final File file = File.createTempFile("HierarchicalDataFileModelTest",
 				".nxs");
 		file.delete();
 		try {
 
 			// Create a file and verify it
-			IHierarchicalDataFile writer = HierarchicalDataFactory.getWriter(file.getAbsolutePath());
-			writer.createDataset(PATH, dtype, shape, buffer, writer.getRoot());
+			NexusFile writer = NexusFileHDF5.createNexusFile(file.getAbsolutePath());
+			Dataset data = DatasetFactory.createFromObject(dtype, buffer, shape);
+			data.setName("Gold");
+			writer.createData(PATH, data, true);
 			writer.close();
-			try (IHierarchicalDataFile reader = HierarchicalDataFactory
-					.getReader(file.getAbsolutePath());) {
+			try (NexusFile reader = NexusFileHDF5.openNexusFile(file.getAbsolutePath())) {
 				printDataContents(reader);
+				reader.close();
 			}
 
 			IHierarchicalDataFileModel model = new HierarchicalDataFileModel(
 					new IHierarchicalDataFileGetReader() {
 
 						@Override
-						public IHierarchicalDataFile getReader()
+						public NexusFile getReader()
 								throws Exception {
 							String absolutePath = file.getAbsolutePath();
-							return HierarchicalDataFactory
-									.getReader(absolutePath);
+							NexusFile nexusFile = NexusFileHDF5.openNexusFile(absolutePath);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "nexusFile". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)